overlord/snapstate/snapmgr.go

```diff
@@ -79,6 +87,8 @@ type SnapSetup struct {
 	Flags SnapSetupFlags `json:"flags,omitempty"`
 
 	SnapPath string `json:"snap-path,omitempty"`
+
+	UpdateInfo *StoreUpdateInfo `json:"update-info,omitempty"`
```

Comment:
this is strange, given that we already have Candidate carrying a SideInfo... what I would do is introduce snap.DownloadInfo (which can be just the urls, I don't think we actually need to send confiment to download), and set Candidate early,  given that DownloadInfo is only for downloads it could even just attached to the task instead of in SnapSetup